test/unit/component/mxgraph/shape/render/utils.test.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2020 Bonitasoft S.A.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { orderActivityMarkers } from '../../../../../../src/component/mxgraph/shape/render/utils';
+import { ShapeBpmnMarkerKind } from '../../../../../../src/model/bpmn/shape';
+
+describe('enforce activity markers order', () => {
+  describe('1 element', () => {
+    it.each(Object.values(ShapeBpmnMarkerKind))(`1 element - %s`, (marker: string) => {
+      const markers = [marker];
+      expect(orderActivityMarkers(markers)).toEqual(markers);
+    });
+  });
+
+  describe('2 elements', () => {
+    it.each([
+      [
+        [ShapeBpmnMarkerKind.EXPAND, ShapeBpmnMarkerKind.LOOP],
```

Comment:
@csouchet I currently don't check all elements order. We could generate all combinations for a given set of markers to ensure the order is always the same.
Do you think it's worth it?
For now, I only check that `[ShapeBpmnMarkerKind.EXPAND, ShapeBpmnMarkerKind.LOOP]` and ` [ShapeBpmnMarkerKind.LOOP, ShapeBpmnMarkerKind.EXPAND]` produce the same result.

Reply:
I think it's better to check all combinations, like we do for the label bounds/font 😉 
Really, it's change nothing, but it's cover all cases.

Reply:
ok, I am on it

Reply:
@csouchet I have just added some permutations. There is still refactoring work to do and more test cases must be added for markers with 3 elements.